s2spy/calendar_shifter.py

```diff
@@ -0,0 +1,169 @@
+            ]
+        )
+    """
+    calendar_shifted = copy.deepcopy(calendar)
+    calendar_shifted.targets[0].gap = _gap_shift(calendar.targets[0], shift)
+    calendar_shifted.precursors[0].gap = _gap_shift(calendar.precursors[0], shift)
+
+    return calendar_shifted
+
+
+def staggered_calendar(
+    calendar: s2spy.time.Calendar, shift: Union[str, dict], n_shifts: int
+) -> List[s2spy.time.Calendar]:
+    """
+    Shift a Calendar instance by a given time offset n times to create a list of shifted
+    calendars. We call this list a staggered calendar.
+    Args:
+        calendar: an s2spy.time.Calendar instance
+        shift: a pandas-like
+            frequency string (e.g. "10d", "2W", or "3M"), or a pandas.DateOffset
+            compatible dictionary such as {days=10}, {weeks=2}, or {months=1, weeks=2}
+        n_shifts: strictly positive integer for the number of shifts
+    Example:
+        Shift an input calendar n times by a given dateoffset and return a list of these
+        shifted calendars.
+        >>> import s2spy.time
+        >>> cal = s2spy.time.Calendar(anchor="07-01")
+        >>> cal.add_intervals("target", "7d")
+        >>> cal.add_intervals("precursor", "7d", gap="14d")
+        >>> cal.add_intervals("precursor", "7d", n=3)
+        >>> cal_shifted = cal_stagger(cal, "7d", 1)
+        >>> cal_shifted
+        [Calendar(
+            anchor='07-01',
+            allow_overlap=False,
+            mapping=None,
+            intervals=[
+                Interval(role='target', length='7d', gap='0d'),
+                Interval(role='precursor', length='7d', gap='14d'),
+                Interval(role='precursor', length='7d', gap='0d'),
+                Interval(role='precursor', length='7d', gap='0d'),
+                Interval(role='precursor', length='7d', gap='0d')
+            ]
+        ),
+        Calendar(
+            anchor='07-01',
+            allow_overlap=False,
+            mapping=None,
+            intervals=[
+                Interval(role='target', length='7d', gap={'days': 7}),
+                Interval(role='precursor', length='7d', gap={'days': 7}),
+                Interval(role='precursor', length='7d', gap='0d'),
+                Interval(role='precursor', length='7d', gap='0d'),
+                Interval(role='precursor', length='7d', gap='0d')
+            ]
+        )]
+    """
+    if not isinstance(n_shifts, int):
+        raise ValueError(
+            "Please input an 'int' type for the 'n' argument."
+            f" Not a {type(n_shifts)}."
+        )
+    if n_shifts <= 0:
+        raise ValueError(
+            "The number of shifts 'n' has to be 1 or greater, " f"not '{n_shifts}'."
+        )
+    cal_staggered = [calendar]
+    for _ in range(n_shifts):
+        cal_shifted = calendar_shifter(cal_staggered[-1], shift)
+        cal_staggered.append(cal_shifted)
+
+    return cal_staggered
+
+
+def calendar_list_resampler(cal_list: list, ds: xr.Dataset) -> xr.Dataset:
```

Comment:
A dim_name kwarg here would be nice. This would give users a bit more freedom. With a default of "step".